driver/normalizer/normalizer.go

```diff
@@ -73,118 +145,183 @@ var Normalizers = []Mapping{
 	// Args in C can have type but be empty (typically in headers, but also in implementations): int main(int, char**)
```

Comment:
It's not only for functions. If I removed this then some `CPPASTName`s wouldn't be promoted to `uast:Identifier` because there are other empty instances of `CPPASTName`s in non semantic objects like enums and other cases. I don't think an empty identifier is against the specs. Or is there any way to promote all empty `CPPASTName's to a nil in a way that the `nil` var will be used everywhere the previous value was expected?

Reply:
I guess mapping from a node to `Is(nil)` should work well. But it will leave nil nodes in arrays (if Identifiers ever happen to be in arrays).

Reply:
You mean like?

```go
	Map(Obj{
		"IASTClass": String("CPPASTName"),
		"Name":      String(""),
	}, Obj{
		uast.KeyType: String("uast:Identifier"),
		"Name":       Is(nil),
	}),
```

Reply:
Second option seems to work well and doesn't break anything _except_ one seemingly unrelated annotation in one fixture removing the roles :-? Investigating it.

Reply:
Fixed :)